src/blog/2024/06/calling-python-script-from-node-red.md

```diff
@@ -0,0 +1,114 @@
+---
+title: Calling a Python script from Node-RED
+subtitle: Guide on how to execute Python scripts from Node-RED
+description: Learn how to seamlessly execute a Python script from Node-RED for advanced data processing and analysis.
+date: 2024-06-21
+authors: ["sumit-shinde"]
+image: /blog/2024/06/images/calling-python-script-from-node-red.png
+tags:
+   - post
+   - node-red
+   - how-to
+   - python
+---
+
+Python's robust data processing capabilities and extensive libraries are well-known in programming. When combined with Node-RED, these technologies can synergize to elevate data analytics and automation to new heights.
+
+This guide walks you through seamlessly integrating Python scripts with Node-RED. You'll gain practical insights, troubleshooting tips, and effective techniques for executing scripts, enabling you to leverage this powerful combination for your IoT projects.
+
+<!--more-->
+
+## Why use python with Node-RED
+
+Integrating Python and Node-RED can significantly enhance your IoT and automation initiatives by leveraging their distinct strengths. Node-RED excels in creating easy workflows, efficiently processing data streams, and integrating hardware, APIs, and. Meanwhile, Python offers a rich set of libraries for advanced tasks such as machine learning and AI, pivotal in realizing Industry 4.0 concepts.
+
+This combination allows developers to build robust and flexible solutions. For instance, while Node-RED manages data flow and device communication, Python can perform complex analytics, and predictive modeling, or integrate with AI frameworks. This integration bridges the gap between data collection and actionable insights, enabling systems to make informed decisions autonomously.
+
+## Installing Python
+
+When executing Python scripts, it's essential to have the Python runtime installed on your system. Since Python is a prerequisite for npm, you might already have it installed if you installed Node-RED via npm 
+
+To verify if Python is installed, open your terminal and execute:
+
+```bash
+python --version
+```
+
+!["Screenshot of terminal showing the python version installed, conforming it is installed"](./images/calling-python-script-from-node-red-py-conformation.png "Screenshot of terminal showing the python version installed, conforming it is installed"){data-zoomable}
+
+The above command displays the version of Python installed on your system as shown in the above image. If the above command doesn't work, try:
+
+```bash
+python3 --version
+```
+
+The specific command to use depends on how Python was installed and configured on your system. However, make sure to use `python <filename>.py` if the first command works, or `python3 <filename>.py` if the second command works, while executing Python scripts.
```

Comment:
AFAIK there's a symlink from `python` in the $PATH from `python3`. And so these are the same executable these days.

Reply:
Yes, in my Windows environment, `python` works, whereas in the Ubuntu VM, I have to use `python3`. This might be because I followed incorrect instructions during the installation in the VM, but keeping this section can help readers if they had done the same before.